README.md

```diff
@@ -118,3 +115,122 @@ suspend fun logout() {
     }
 }
 ```
+
+#### Error handling
+By default, access token is considered as expired when server returns *HTTP 401 Unauthorized* response. Refresh token is considered as expired when server returns *HTTP 400 Bad Request* or *HTTP 401 Unauthorized* responses. You can modify this behavior with providing custom `AuthErrorChecker` to `OAuthManager`.
+```kotlin
+// Equivalent to invalid access token in OAuth2 flow - called when network request failed
+override fun invalidCredentials(t: Throwable): Boolean {
+	if (t is HttpException) {
+		if (t.code() == HTTP_UNAUTHORIZED) {
+			return true // access token invalid
+		}
+    }
+    return false // access token valid
+}
+
+// Equivalent to invalid refresh toke in OAuth2 flow - called when request for credentials refresh failed
+override fun invalidRefreshCredentials(t: Throwable): Boolean {
+    if (t is HttpException) {
+	    when (t.code()) {
+		    HTTP_BAD_REQUEST, HTTP_UNAUTHORIZED -> true // refresh token invalid
+		    else -> false // refresh token valid
+		}
+    }
+    return false // refresh token valid
+}
+```
+
+### Usage - custom authentication flow
+When using different auth flow than OAuth2, you have to write custom `AuthManager` implementation. Otherwise usage is exactly the same as when using provided `OAuthManager` .
+
+But don't worry, it's easier than it sounds! These are necessary steps for the implementation:
+
+1. Create custom `AuthCredentials`
+2. Implement `AuthStore` for storing your credentials
+3. Create OkHttp `Interceptor` for injecting authorization metadata to network requests
+4. Create your custom `AuthManager<C : AuthCredentials>`
+
+#### AuthCredentials
+Simple interface for auth credentials model entity. If your credentials contains information about time of expiration, you should override `areExpired()` function.
+
+Example:
+```kotlin
+data class OAuthCredentials(
+    val accessToken: String,
+	val refreshToken: String,
```

Comment:
here is a weird mix of tabs/spaces in this README. I would go for spaces only :D

Reply:
Fixed in 905302a418995e9e8c0cfc0b5c2de152c589879a